src/CFamily.UnitTests/CMake/ActiveConfigProviderTests.cs

```diff
@@ -0,0 +1,120 @@
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("rootDirectory");
+        }
+
+        [TestMethod]
+        public void Get_MissingSettingsFile_ReturnsDefaultConfig()
+        {
+            const string rootDir = "c:\\aaa";
+            var fullSettingsPath = CalcFullSettingsPath(rootDir);
+
+            var fileSystem = CreateFileSystemWithNoFiles();
+            var logger = new TestLogger(logToConsole: true);
+            var testSubject = new ActiveConfigProvider(logger, fileSystem.Object);
+
+            testSubject.GetActiveConfig(rootDir).Should().Be(ExpectedDefaultConfig);
+
+            AssertSettingsFileExistenceChecked(fileSystem, fullSettingsPath);
+        }
+
+        [TestMethod]
+        [DataRow("not valid json", ExpectedDefaultConfig)]
+        [DataRow("{ 'SettingIsMissing' : 'aaa'}", ExpectedDefaultConfig)]
+        [DataRow("{ 'CurrentProjectSetting': 'my config' }", "my config")]
+        public void Get_FileExists_ReturnsCorrectConfig(string settingsJson, string expectedConfig)
+        {
+            const string rootDir = "c:\\xxx";
+            var fullSettingsPath = CalcFullSettingsPath(rootDir);
+
+            var fileSystem = CreateFileSystemWithFile(fullSettingsPath, settingsJson);
+            var logger = new TestLogger(logToConsole: true);
+            var testSubject = new ActiveConfigProvider(logger, fileSystem.Object);
+
+            testSubject.GetActiveConfig(rootDir).Should().Be(expectedConfig);
+
+            AssertSettingsFileExistenceChecked(fileSystem, fullSettingsPath);
+            AssertSettingsFileRead(fileSystem, fullSettingsPath);
+        }
+
+        private static Mock<IFileSystem> CreateFileSystemWithNoFiles()
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(It.IsAny<string>())).Returns(false);
+            return fileSystem;
+        }
+
+        private static Mock<IFileSystem> CreateFileSystemWithFile(string fullPath, string content)
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(fullPath)).Returns(true);
+            fileSystem.Setup(x => x.File.ReadAllText(fullPath)).Returns(content);
+            return fileSystem;
+        }
+
+        private static void AssertSettingsFileExistenceChecked(Mock<IFileSystem> fileSystem, string fullSettingsPath)
+        {
+            fileSystem.Verify(x => x.File.Exists(fullSettingsPath), Times.Once);
+        }
+
+        private static void AssertSettingsFileRead(Mock<IFileSystem> fileSystem, string rootDirectory)
+        {
+            var fullPath = CalcFullSettingsPath(rootDirectory);
+            fileSystem.Verify(x => x.File.ReadAllText(rootDirectory), Times.Once);
+        }
+
+        private static string CalcFullSettingsPath(string rootDirectory) =>
+            Path.Combine(rootDirectory, ".vs", "ProjectSettings.json");
```

Comment:
I think it's worth making RelativeSettingsPath internal in the product code and re-using it.

Reply:
(same comment as for the `ExpectedDefaultConfig` constant above)